### PR TITLE
Initial scope of the project

### PR DESCRIPTION
**Pull request by otaviojava**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
-Spring Boot maven with MySQL example configuration for Platform.sh. Maintained by the Platform.sh team.
+# Spring Boot maven with MySQL for Platform.sh
+
+This project provides a starter kit for Spring Boot maven with MySQL projects hosted on [Platform.sh](http://platform.sh).
+
+We include recommended .platform.app.yaml and .platform files that should suffice for most use cases. You are free to tweak them as needed for your particular site.
+
+## Platform.sh configuration files
+
+* [`scripts/platformsh`](/scripts/platformsh) - This directory contains our update script to keep this repository in sync with the Drupal Composer project.  It may be safely ignored or removed.
```
[the remark this answers is not in the document]
Comment (otaviojava) [reply README.md]: Thank you, I removed this line.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,21 @@
...
+* [.platform/routes.yaml](https://docs.platform.sh/configuration/routes.html): Platform.sh allows you to define the routes.
```
[the remark this answers is not in the document]
Comment (otaviojava) [reply README.md]: I don't get it. 
Do you mean?

[the remark this answers is not in the document]
Comment (chadwcarlson) [reply README.md]: I meant that you have code formatting on for `scripts/platformsh`, but not for .platform/routes.yaml, .platform/services.yaml, and .platform.app.yaml, which now are just text.

[the remark this answers is not in the document]
Comment (Crell) [reply README.md]: All file names and path names should have backticks around them to set them off as code in Markdown.  That's what Chad's referring to.

[the remark this answers is not in the document]
Comment (otaviojava) [reply README.md]: Done, thank you.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,21 @@
-# template-spring-boot-maven-mysql
-Spring Boot maven with MySQL example configuration for Platform.sh. Maintained by the Platform.sh team.
+# Spring Boot maven with MySQL for Platform.sh
```
[the remark this answers is not in the document]
Comment (otaviojava) [reply README.md]: Thank you.

File: .platform.app.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+name: app
```
Comment (Crell) [inline .platform.app.yaml]: This file should be well-commented, as it's potentially someone's first time seeing this file.  (Not all of our templates are good about that right now, but we should fix that.)  Just steal the comments from one of the better-commented examples outright.

File: src/main/java/sh/platform/template/PersonRepository.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package sh.platform.template;
+
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+public interface PersonRepository extends PagingAndSortingRepository<Person, Long> {
```
Comment (Crell) [inline src/main/java/sh/platform/template/PersonRepository.java]: I take it PagingAndSortingRepository is a fully functional implementation already, which is why this class is empty?

File: src/main/java/sh/platform/template/DataSourceConfig.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import javax.sql.DataSource;
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean(name = "dataSource")
+    public DataSource getDataSource() {
+        Config config = new Config();
+        MySQL database = config.getCredential("database", MySQL::new);
+        return database.get();
```
Comment (Crell) [inline src/main/java/sh/platform/template/DataSourceConfig.java]: To my PHP-trained mind, this is a hard-coded class that isn't using dependency injection and thus is Bad(tm).  Is this actually typical Java design?  Doesn't that make the data source hard-coded and hard-to-test in ways you don't want?

File: src/main/java/sh/platform/template/PersonRepository.java (hunk 5 above)
Comment (otaviojava) [reply src/main/java/sh/platform/template/PersonRepository.java]: Spring will implement it from my dependencies, it is magic, isn't?

File: src/main/java/sh/platform/template/DataSourceConfig.java (hunk 6 above)
Comment (otaviojava) [reply src/main/java/sh/platform/template/DataSourceConfig.java]: Yes, that point is what makes hard to select environments by Java :(
Usually, we use some property file, therefore, I can have one to run locally, to run tests and to production.

**Comment by chadwcarlson** [comment]
This all runs fine and looks good to me. The only feedback I could give on both repos, especially since my eye is not sensitive to Java yet, is some in-line comments. I think they would be helpful, especially since they will be used as public facing templates.

**Pull request merged**